pinot-spi/src/main/java/org/apache/pinot/spi/utils/TimestampUtils.java

```diff
@@ -19,19 +19,27 @@
 package org.apache.pinot.spi.utils;
 
 import java.sql.Timestamp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
 
 
 public class TimestampUtils {
+
+  private static final String[] SDF_FORMATS = {
```

Comment:
We can pre-compile these formats to `SimpleDateFormat` to avoid compiling them for each method call

Reply:
Unless I’m mistaken, SDF is not thread safe which is why it was common to store them in thread locals pre ZDK8. Using DateTimeFormatter should help if you’re concerned about performance (but note that 2 exceptions have already been thrown by now) because they can be shared.

Reply:
You are correct. Per the documentation: 
```
 * Date formats are not synchronized.
 * It is recommended to create separate format instances for each thread.
 * If multiple threads access a format concurrently, it must be synchronized
 * externally.
```